api/google-manager/services/google-manager.js

```diff
@@ -171,56 +172,73 @@ class GoogleManager {
         auth: this.calendarAuth
       });
 
-      const createdEvent = await calendar.events.insert({
-        calendarId,
-        conferenceDataVersion: 1,
-        requestBody: {
-          end: {
-            date: this.getCurrentDate(),
-          },
-          start: {
-            date: this.getCurrentDate(),
-          },
-          recurrence: [
-            'RRULE:FREQ=DAILY'
-          ],
-          attendees: [
-            {
-              email: groupEmail,
-              responseStatus: 'accepted'
-            }
-          ],
-          conferenceData: {
-            createRequest: {
-              requestId: uuidv4(),
-              conferenceSolutionKey: {
-                type: 'hangoutsMeet',
-              },
-            }
-          },
-          summary: `${title} Common Room`,
-          //This prevents the event from blocking time on the calendar
-          transparency: 'transparent'
+      // get the meeting newMeetingTimes
+      const project = await strapi.query('project').findOne({ id: projectID });
+
+      const newMeetingTimes = project.newMeetingTimes;
+       
+      await Promise.all(newMeetingTimes.map(async (meetingTime) => {
+        const startTime = new Date(meetingTime.startTime);
+        const endTime = new Date(meetingTime.endTime);
+        const timeZone = meetingTime.IANA_timeZone;
+        // invalid timezone
+        if (!timezoneSet.has(timeZone)) {
+          throw 'invalid timeZone!';
```

Comment:
We should return an error instead of throwing an exception.

Reply:
How can I return an error? I used "return new Error("This is not an IANA timeZone!")" but it didn't work. If I used "throw new Error("This is not an IANA timeZone!")", it seems to work.

Reply:
@alearm246 do you know how we can return something HTTP Bad Request to the caller of this API? Or maybe we can't do that because this is not part of the lifecycle hook?

Reply:
@chungthuang @chungthuang she could throw an exception and wherever this method is being called she can catch it and use strapi's ctx.badRequest() function. Unless you mean something else.

Reply:
Do you mean using ctx.badRequest() after line 186? But what is ctx in this function?

Reply:
@jiawen195 My idea was that you would throw an exception in this method and wherever you use this method you would catch the exception using a try catch block in one of strapi's functions that provide the ctx object.

Reply:
I recommend https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Control_flow_and_error_handling#exception_handling_statements reading for error handling.

Reply:
From https://forum.strapi.io/t/throw-error-message-in-lifecycle-hook-doesnt-change-the-error-message-display-in-the-ui/1515, it looks like you should return `strapi.errors.badRequest`.